lib/Sema/DerivedConformanceCodable.cpp

```diff
@@ -89,6 +89,12 @@ static CodableConformanceType typeConformsToCodable(TypeChecker &tc,
   // Some generic types need to be introspected to get at their "true" Codable
   // conformance.
   auto canType = target->getCanonicalType();
+  if (auto referenceType = dyn_cast<ReferenceStorageType>(canType)) {
```

Comment:
No, that's fair — I figured that's what you were thinking of. That version is called `encodeWeak()` at the moment, though I'm thinking we might want to rename that back to `encodeConditional()`. Regardless, it might be appropriate to use that form for non-`strong` properties, but for now, we encode unconditionally, which will always work.